### ISSUE TITLE
Enhancement: [no-floating-promises] add ignorePattern

### ISSUE THREAD

**Issue by kkmuffme** [typescript-eslint/typescript-eslint#7006]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-floating-promises/

### Description

When moving to TS/upgrading legacy code you often have e.g. jQuery.ajax() calls which now return a promise, but 99% of people write their success/error callbacks directly in the param object.

This leads to tons of eslint-disable-next-line comments.

**An option for the rule like `ignorePattern` (which exists in eslint for various rules) which would exclude certain patterns from reporting an error would be great**

### Fail

```typescript
no change
```

### Pass

```typescript
jQuery.ajax({
	method: 'POST',
	url: 'example.com',
	data: {
		foo: 'bar',
	},
	complete: function() {
		doSomething();
	}
});
```

**Comment by JoshuaKGoldberg** [comment]
> 99% of people write their success/error callbacks directly in the param object.

The rule is generally right to warn on that pattern. There's often no guarantee available in the type system that those calls don't themselves throw an error:

```ts
jQuery.ajax({
	complete: doSomething,
	method: 'GET',
	url: 'example.com',
});

function doSomething() {
  if (somethingRiskyThatOccasionallyFails()) {
    throw new Error("oh no");
  }
}
```

...which is why `complete` methods and similar on jQuery Ajax calls are not how I would recommend using them. Even the [jQuery.ajax() docs](https://api.jquery.com/jquery.ajax/) don't include an explicit example of them, as they're not the primary recommended way of using the functions.

> moving to TS/upgrading legacy code

In general, we try not to add options for old/legacy patterns that the community has moved away from. We'd want to see an example of a case where a "floating" Promise is a _good_ way of writing the code. jQuery's `complete` is not convincing.

---

That being said, we have seem more convincing cases, such as #5231. I'll re-file this issue with a more broad context and links to older issues.

(apologies for hijacking your issue - I've had bad experiences before from discussions on reasonable feature requests getting sidetracked by non-optimal starting suggestions)
